pathd/path_pcep_controller.c

```diff
@@ -635,8 +683,30 @@ int pcep_thread_event_handler(struct thread *thread)
 		ret = pcep_thread_event_update_pce_options(ctrl_state, pcc_id,
 							   pce_opts);
 		break;
+	case EV_UPDATE_BEST_PCE:
+		ret = pcep_thread_event_update_best_pce(ctrl_state, pcc_id);
+		break;
+	case EV_UPDATE_TIMER_BEST_PCE:
+		if (!pcc_id) {
+			// resync whatever were new best
+			pcc_id = calculate_best_pce(ctrl_state);
+			if (pcc_id) {
+				struct pcc_state *pcc_state =
+					get_pcc_state(ctrl_state, pcc_id);
+				pcep_thread_schedule_sync_best_pce(
+					ctrl_state, pcc_id, 30 /*delay*/,
+					&pcc_state->t_reconnect);
+			}
+		}
+		break;
 	case EV_REMOVE_PCC:
-		ret = pcep_thread_event_remove_pcc(ctrl_state, pcc_id);
+		pce_opts = (struct pce_opts *)payload;
+		ret = pcep_thread_event_remove_pcc(ctrl_state, pce_opts);
+		new_best_pcc_id = calculate_best_pce(ctrl_state);
+		if (new_best_pcc_id) {
+			ret = pcep_thread_event_update_best_pce(
```

Comment:
This probably dosen't need an event either. And if there is multiple places doing the calculate and notify pccs, it should be done in its own function.